File: 1. Creational Design Patterns/Prototype/Java/PrototypeTemplate.java
// Prototype Pattern – Java Template
interface Prototype<T> { T cloneProto(); }
class ConcretePrototype implements Prototype<ConcretePrototype> {
    int data;
    ConcretePrototype(int d){ data=d; }
    public ConcretePrototype cloneProto(){ return new ConcretePrototype(this.data); }
}
public class PrototypeTemplate { public static void main(String[] args){ ConcretePrototype p1=new ConcretePrototype(42); ConcretePrototype p2=p1.cloneProto(); } }

File: 1. Creational Design Patterns/Prototype/Kotlin/PrototypeTemplate.kt
// Prototype Pattern – Kotlin Template
interface Prototype<T> { fun cloneProto(): T }
class ConcretePrototype(var data: Int) : Prototype<ConcretePrototype> {
    override fun cloneProto() = ConcretePrototype(data)
}
fun main() {
    val original = ConcretePrototype(42)
    val copy = original.cloneProto()
}
